src/components/anchor-navigation/anchor-navigation.component.js

```diff
@@ -0,0 +1,133 @@
+    const indexOfSmallestNegativeTopOffset = topEdgeOffsets.reduce((currentTopIndex, topOffset, index) => {
+      if (topOffset > navTopOffset) return currentTopIndex;
+      return topOffset > topEdgeOffsets[currentTopIndex] ? index : currentTopIndex;
+    }, 0);
+
+    setSelectedIndex(indexOfSmallestNegativeTopOffset);
+  }, 200), []);
+
+  useEffect(() => {
+    window.addEventListener('scroll', setSelectedAnchorBasedOnScroll, true);
+    return () => window.removeEventListener('scroll', setSelectedAnchorBasedOnScroll, true);
+  }, [setSelectedAnchorBasedOnScroll]);
+
+  const scrollToItem = (event, index) => {
+    event.preventDefault();
+    const sectionToScroll = sectionRefs.current[index].current;
+    sectionToScroll.scrollIntoView({ block: 'start', inline: 'nearest', behavior: 'smooth' });
+
+    // eslint-disable-next-line max-len
+    const defaultFocusableSelectors = 'button, [href], input:not([type="hidden"]), select, textarea, [tabindex]:not([tabindex="-1"])';
+    const firstFocusableElement = sectionToScroll.querySelector(defaultFocusableSelectors);
+    if (firstFocusableElement) {
+      firstFocusableElement.focus({ preventScroll: true });
+    }
+  };
+
+  const focusNavItem = (event, index) => {
+    event.preventDefault();
+    let newIndex = index;
+    if (index === -1) {
+      newIndex = anchorRefs.current.length - 1;
+    } else if (index === anchorRefs.current.length) {
+      newIndex = 0;
+    }
+    anchorRefs.current[newIndex].current.focus();
+  };
+
+  const handleKeyDown = (event, index) => {
+    if (Event.isUpKey(event)) {
+      focusNavItem(event, index - 1);
+    } else if (Event.isDownKey(event)) {
+      focusNavItem(event, index + 1);
+    } else if (Event.isEnterKey(event) || Event.isSpaceKey(event)) {
+      scrollToItem(event, index);
+    }
+  };
+
+  return (
+    <StyledAnchorNavigation
+      ref={ contentRef }
+      data-component='anchor-navigation'
+      styleOverride={ styleOverride.root }
+    >
+      <StyledNavigation
+        ref={ navigationRef }
+        data-element='anchor-sticky-navigation'
+        styleOverride={ styleOverride.navigation }
+      >
+        {React.Children.map(
+          stickyNavigation.props.children,
+          (child, index) => React.cloneElement(child, {
+            isSelected: index === selectedIndex,
+            tabIndex: index === selectedIndex ? 0 : -1,
+            onClick: ev => scrollToItem(ev, index),
+            onKeyDown: ev => handleKeyDown(ev, index),
+            ref: anchorRefs.current[index]
+          })
+        )}
+      </StyledNavigation>
+      <StyledContent styleOverride={ styleOverride.content }>
+        {children}
+      </StyledContent>
+    </StyledAnchorNavigation>
+  );
+};
+
+AnchorNavigation.propTypes = {
+  children: PropTypes.node,
+  /** The AnchorNavigationItems components to be rendered in the sticky navigation */
+  stickyNavigation: (props, propName, componentName) => {
```

Comment:
Unless you need that custom error message, maybe we can consider implementing shorter version of children.
```
children: PropTypes.shape({
    type: PropTypes.oneOf([AnchorNavigationItem, PropTypes.element])
  }).isRequired
```

Reply:
Problem is that I'm checking the children of the <> (React.fragment) component here, so I'm unable to check it with PropTypes.shape here